File: BudgetTracker.Server/Models/Transaction.cs
using System.ComponentModel.DataAnnotations;

namespace BudgetTracker.Server.Models
{
    public class Transaction
    {
        public int Id { get; set; }
        public required int Type { get; set; } // "expense" = 0, "income" = 1, "transfer" = 2
        public int Amount { get; set; } = 0;
        public required int AccountId { get; set; }
        [MaxLength(255)]
        public string? AccountName { get; set; }
        public required DateTime? Date { get; set; }
        public int? CategoryId { get; set; }
        [MaxLength(255)]
        public string? CategoryName { get; set; }
        [MaxLength(50)]
        public string? CategoryIcon { get; set; }
        public int? AccountIdTo { get; set; }
        [MaxLength(255)]
        public string? AccountNameTo { get; set; }
        public string? Note { get; set; }
        [MaxLength(450)]
        public string? Owner { get; set; } // User.Id
        public DateTime Changed { get; set; }
    }
}

File: budgettracker.client/src/app/shared/models/transaction.ts
import {OperationTypes, Statuses} from "./enums";

export interface Transaction {
  id: number;
  type: OperationTypes; // "expense" = 0, "income" = 1, "transfer" = 2
  amount: number;
  accountId: number;
  accountName: string;
  categoryId: number;
  categoryName: string;
  categoryIcon: string;
  date: Date;
  accountIdTo: number;
  accountNameTo: string;
  note: string;
  status: Statuses; // "Active" = 1, "Inactive" = 0
  owner: string; // User.Id
  changed: Date;
}

export interface MonobankTransaction {
  id: string;
  accountId?: string;
  time: number;
  description?: string;
  mcc?: number;
  originalMcc?: number;
  hold?: boolean;
  amount: number;
  operationAmount: number;
  currencyCode: number;
  commissionRate?: number;
  cashbackAmount?: number;
  balance?: number;
  comment?: string;
  receiptId?: string;
  invoiceId?: string;
  counterEdrpou?: string;
  counterIban?: string;
  counterName?: string;
  owner?: string; // User.Id
}

export interface TransactionHttpParams {
  startDate?: Date,
  endDate?: Date
}
